simple_history/management/commands/_populate_utils.py

```diff
@@ -15,16 +16,57 @@ def get_history_model_for_model(model):
     return getattr(model, manager_name).model
 
 
-def bulk_history_create(model, history_model, batch_size):
-    """Save a copy of all instances to the historical model."""
-    historical_instances = [
-        history_model(
-            history_date=getattr(instance, '_history_date', now()),
-            history_user=getattr(instance, '_history_user', None),
-            **{
-                field.attname: getattr(instance, field.attname)
-                for field in instance._meta.fields
-            }
-        ) for instance in model.objects.all()]
-    history_model.objects.bulk_create(historical_instances,
-                                      batch_size=batch_size)
+def bulk_history_create(
```

Comment:
This isn't transaction-safe. I don't love the idea of wrapping this in a transaction either, since it may block for an unacceptable amount of time depending on table size. Maybe add an option to wrap this in a transaction? Or at least document that this isn't transaction-safe.

Reply:
why is it now not transaction safe?

Reply:
The `bulk_create` itself is wrapped in an atomic transaction, but you're calling a series of `bulk_create` calls that aren't all together wrapped in a transaction. So if one of those bulk_create calls fails, then you'll have some of your data inserted in the database, rather than rolling it back so that none of your data is inserted.

Reply:
gotcha. thanks.